### PR TITLE
log-backup: remove checkpoint V2 from codebase

### PR DESCRIPTION
**Pull request by YuJuncen**
Signed-off-by: Yu Juncen <yujuncen@pingcap.com>

### What is changed and how it works?

Issue Number: Ref #13196

What's Changed:

This PR adapt the integration test cases for removing checkpoint v2.

```commit-message
Now, all integration test cases uses checkpoint V3.
Removed `test_inflight_message` because it is invalid in V3.
```

### Check List 

Tests 

- No code (Who tests tests?)

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by joccau** [comment]
pub enum CheckpointProvider {
    Store(u64),
    Region { id: u64, version: u64 },
    Task,
    Global,
}
Can we remove the CheckpointProvider::Store and CheckpointProvider::Region.
https://github.com/tikv/tikv/blob/c48b0cfa8630d949a6e32a14a74fad7fdbc78c02/components/backup-stream/src/metadata/client.rs#L76

File: src/config.rs [hunk 1]
```diff
@@ -2496,9 +2496,6 @@ pub struct BackupStreamConfig {
     pub initial_scan_pending_memory_quota: ReadableSize,
     #[online_config(skip)]
     pub initial_scan_rate_limit: ReadableSize,
-    #[serde(skip)]
-    #[online_config(skip)]
-    pub use_checkpoint_v3: bool,
```
Comment (overvenus) [inline src/config.rs:2501]: Will it cause compatibility issue?

Comment (YuJuncen) [reply src/config.rs:2501]: In all release version this field should be `#[serde(skip)]`, I guess this would be fine.🤔

**Review by overvenus — APPROVED** [review]

**Review by joccau — APPROVED** [review]
LGTM

**Review by 3pointer — APPROVED** [review]

**Comment by YuJuncen** [comment]
test_prevote_reboot_minority_followers 🤔
/run-test

**Comment by YuJuncen** [comment]
test_read
/run-test

**Comment by YuJuncen** [comment]
operation::query::local::tests::test_read
/run-test

**Pull request merged**